lib/db.js

```diff
@@ -239,39 +246,71 @@ class LimitDBRedis extends EventEmitter {
           count *= (bucketKeyConfig.skip_n_calls + 1);
         }
       }
-
-
     }
 
-    this.redis.take(key,
-      bucketKeyConfig.ms_per_interval || 0,
-      bucketKeyConfig.size,
-      count,
-      Math.ceil(bucketKeyConfig.ttl || this.globalTTL),
-      bucketKeyConfig.drip_interval || 0,
-      (err, results) => {
-        if (err) {
-          return callback(err);
-        }
-
-        const remaining = parseInt(results[0], 10);
-        const conformant = parseInt(results[1], 10) ? true : false;
-        const currentMS = parseInt(results[2], 10);
-        const reset = parseInt(results[3], 10);
-        const res = {
-          conformant,
-          remaining,
-          reset: Math.ceil(reset / 1000),
-          limit: bucketKeyConfig.size,
-          delayed: false,
-        };
-
-        if (bucketKeyConfig.skip_n_calls > 0) {
-          this.callCounts.set(key, { res, count: 0 });
-        }
-
-        return callback(null, res);
-      });
+    if (bucket.elevated_limits && params.allowERL===true) {
```

Comment:
should we fail or return an error if one is true but the other false? Is this a valid case or would look like a misconfiguration? Because if it's a misconfiguration we would be silently doing normal `take` and client wouldn't notice at all (until they hit limits and are not elevated at least).

Reply:
- `bucket.elevated_limits` presence means that the bucket has a configuration that should be used if/when ERL is activated.
- `allowERL=true` means, that the caller wants to allow the new ERL mechanism to be used, "if necessary".

It's not an error if `allowERL=true` for a bucket that doesn't have `elevated_limits` configured. Because `allowERL` is to allow the use of ERL for cases when it makes sense, which are
- if bucket has `elevated_limits` configured and has run out of tokens
- bucket was in ERL mode already, and `activation_period` has still not been expired.
- (in the next PR) bucket has `elevated_limits.erlEnabled=true`.

If the case is none of those, it'll just use the normal flow.

It's not an error if bucket has `elevated_limits` configured while `allowERL=false`, because this represents the case when ERL is disabled as a breaking-glass mechanism.